pkgs/desktops/gnome/extensions/update-extensions.py

```diff
@@ -269,7 +274,7 @@ def scrape_extensions_index() -> List[Dict[str, Any]]:
         f"Done. Writing results to extensions.json ({len(processed_extensions)} extensions in total)"
     )
 
-    with open("extensions.json", "w") as out:
+    with open(os.path.join(os.path.dirname(os.path.realpath(__file__)), "extensions.json"), "w") as out:
```

Comment:
Maybe factor out the entire expression into a variable, since it's now sufficiently complex and used three times in the code.

Reply:
I think variable indeed makes more sense than a repeated function call. Also you might want to consider [`pathlib.Path`](https://docs.python.org/3/library/pathlib.html#basic-use)